### PR TITLE
Add OpenShift SecurityContextConstraints object linting

### LINKED ISSUE #649: [FEATURE_REQUEST] Add OpenShift SecurityContextConstraints
**Description of the problem/feature request**
Openshift uses SecurityContextConstraints to provide additional capabilities for Kubernetes resources. Analysis of the privileges in this resource is important from security perspective.

**Description of the existing behavior vs. expected behavior**
Currently SecurityContextConstraints objects not supported

### PR DESCRIPTION
**Pull request by mancubus77**
resolves #649

Add SecurityContextConstraints objects parsing, and adds check of `allowPrivilegedContainer: true` what enabled privileged mode to attached objects.
```
Name: scc-deny-privileged-container
Description: Indicates when allowPrivilegedContainer SSC set to True
Remediation: SecurityContextConstraints has AllowPrivilegedContainer set to "true". Using this option is dangerous, please consider using allowedCapabilities instead
Template: scc-deny-privileged-container
Parameters: map[AllowPrivilegedContainer:true]
Enabled by default: false

```

### CODE REVIEW

File: pkg/templates/sccdenypriv/template_test.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+func (s *SCCPrivTestSuite) addSCCWithPriv(name string, allowFlag bool) {
+	s.ctx.AddMockSecurityContextConstraints(s.T(), name, allowFlag)
+}
+
+func (s *SCCPrivTestSuite) TestPrivFalse() {
+	const (
+		acceptableScc = "scc-priv-false"
+	)
```
Comment (janisz) [inline pkg/templates/sccdenypriv/template_test.go:35]:
```suggestion
	const acceptableScc = "scc-priv-false"
```

File: docs/generated/checks.md [hunk 2]
```diff
@@ -513,6 +513,21 @@ key: owner
 **Remediation**: Set runAsUser to a non-zero number and runAsNonRoot to true in your pod or container securityContext. Refer to https://kubernetes.io/docs/tasks/configure-pod-container/security-context/ for details.
 
 **Template**: [run-as-non-root](templates.md#run-as-non-root-user)
+## scc-deny-privileged-container
+
+**Enabled by default**: No
+
+**Description**: Indicates when allowPrivilegedContainer SSC set to True
+
+**Remediation**: SecurityContextConstraints has AllowPrivilegedContainer set to "true". Using this option is dangerous, please consider using allowedCapabilities instead.
```
Comment (janisz) [inline docs/generated/checks.md:522]: Should we add a links to allowedCapabilites?

**Review by janisz — COMMENTED** [review]
Nice, I've just left small nitpics comment

**Comment by mancubus77** [comment]
Thank you @janisz 
Updated according your advice

**Comment by mancubus77** [comment]
👋 @janisz 
Just wondering if you are fine with the commit
Thank you

**Review by janisz — APPROVED** [review]

**Pull request merged**